### DISCUSSION TITLE
Support for Vulkan on Stage3D

### DISCUSSION THREAD

**Discussion by Fancy2209** [airsdk/Adobe-Runtime-Support#3088] [Ideas]
Vulkan seems to perform better than OpenGL on Most devices that support it, so it'd be nice if it could be supported by S3D Some day

**Comment by ajwfrost** [comment]
We'd considered it.. quite a lot of effort (and of course Vulkan doesn't work on some of the older Android devices, and we still have people using those..) - it would have been what we'd most likely have used if Apple ever went through with their threat to remove OpenGL ES support on iPhones (using MoltenVK to run Vulkan APIs using Metal..)

Having just read up on some of their latest developments in portability... if we could really switch to a single API and then have that run cross-platform, it may actually be quite useful! Will see if someone fancies investigating it further here :-)

thanks

**Reply by Fancy2209** [reply]
Yeah, I think one of the main improvements of Vulkan is that it'd work for everything but Web (I believe VK even works on the Nintendo Switch), so I think it'd be useful to have,  maybe it'd make sense to have OpenGL and DX as fallbacks?

**Reply by FliplineStudios** [reply]
As long as switching to Vulkan wouldn't negatively affect GPU render mode -- I know we're not the only developers who rely on GPU render mode for our apps' performance, it'd be a shame if that mode was affected in the process of changing things for direct/Stage3D...

**Reply by ajwfrost** [reply]
We could probably leave that GPU rendering alone for now, they're generally very different mechanisms internally. The Stage3D stuff is pretty stand-alone and is based on what sort of context is requested, so adding a 'vulkan' option should be independent then of other stuff.
